### ISSUE TITLE
test-http2-client-promisify-connect broken on node-chakracore

### ISSUE THREAD

**Issue by mike-kaufman** [nodejs/node-chakracore#395]
ChakraShim's v8::Promise::Resolver::New is not implemented.  See [here](https://github.com/nodejs/node-chakracore/blob/master/deps/chakrashim/src/v8resolver.cc#L26-L30)

Once fixed, we should be able to re-enable `test\parallel\test-http2-client-promisify-connect.js` test.

**Repro:**
1.  Run following code through node-chakracore build

```javascript
const {  createPromise } = process.binding('util');
const p = createPromise();
if (!p) {
  console.log('createPromise() returned undefined');
} else {
  console.log('createPromise() returned something');
}
```

**Expected:**  result of `createPromise()` to return valid value.
**Acutal:**  `createPromise()` returns undefined.

**Comment by MSLaguana** [comment]
This is also impacting more tests (and presumably more code); `test-zlib-empty-buffer.js` uses `util.promisify` and ends up asserting due to the lack of `Promise::Resolver` support.

**Comment by kfarnung** [comment]
The original test was fixed and removed from the skip list in bafa0def68942e34517637d3dd300f57a5d2021c, I'll send out a PR for the other test.

**Comment by kfarnung** [comment]
Resolved in ec7245944a7fbf4a0d3b50a4fcb4b7aa9cc37fa2
